src/inputs/SelectWithAction/SelectWithAction.js

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React from 'react';
+import { FormControl, Select, Divider, makeStyles, MenuItem, ListItemText, ListItemIcon } from '@material-ui/core';
+import CheckIcon from '@material-ui/icons/Check';
+import PropTypes from 'prop-types';
+
+const useStyles = makeStyles((theme) => ({
+  multipleSelect: {
+    width: '100%',
+  },
+  listItemIcon: {
+    minWidth: '44px',
+  },
+}));
+export const SelectWithAction = ({ labels, options, selectedOption, onOptionSelected, isReadOnly, actions }) => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.multipleSelect}>
+      <FormControl variant="outlined" fullWidth={true}>
+        <Select
+          disabled={isReadOnly}
+          value={selectedOption}
+          onChange={onOptionSelected}
+          renderValue={(value) => value}
+          MenuProps={{
+            anchorOrigin: {
+              vertical: 'bottom',
+              horizontal: 'left',
+            },
+            transformOrigin: {
+              vertical: 'top',
+              horizontal: 'left',
+            },
+            getContentAnchorEl: null,
+          }}
+        >
+          {options.map((role) => (
+            <MenuItem key={role} value={role}>
+              <ListItemIcon
+                className={classes.listItemIcon}
+                style={selectedOption === role ? { visibility: 'visible' } : { visibility: 'hidden' }}
+              >
+                <CheckIcon color="secondary" />
+              </ListItemIcon>
+              <ListItemText primary={labels[role] ?? role} />
+            </MenuItem>
+          ))}
+          {actions.length > 0 && <Divider />}
+          <ul>
+            {actions.map((action) => (
+              <MenuItem key={action.id} className={action.classes} value={action.id} onClick={action.onClick}>
+                <ListItemIcon className={classes.listItemIcon} style={{ color: 'inherit' }}>
+                  {action.icon}
+                </ListItemIcon>
+                <ListItemText primary={action.label} />
+              </MenuItem>
+            ))}
+          </ul>
+        </Select>
+      </FormControl>
+    </div>
+  );
+};
+SelectWithAction.propTypes = {
+  options: PropTypes.array.isRequired,
+  selectedOption: PropTypes.string.isRequired,
+  onOptionSelected: PropTypes.func.isRequired,
+  labels: PropTypes.object.isRequired,
+  isReadOnly: PropTypes.bool.isRequired,
```

Comment:
Good idea ! But I'm not sure for labels: we need them to display options (and not always user's roles), what can be generic enough to be default label for options ?

Reply:
You're right, I forgot the roles translations were in the labels :smile: 

Then maybe we could merge the props `labels` and `options` into a single prop `options`, containing for example:
```
{
  myRoleA: 'label for role A',
  myRoleB: 'label for role B',
  myRoleC: 'label for role C'
}
```
and use its keys as "options", and its values as labels (otherwise the `labels` prop would have the shape above, and the `options` prop would only be a repetition of the dict values)

Reply:
Great, thank you :slightly_smiling_face: